File: app/src/main/java/com/rokid/cxrmsamples/activities/model/enity/Pic.java
package com.rokid.cxrmsamples.activities.model.enity;

public class Pic {

    public int id;
    public String path_leaf;
    public String path_seg;
    public float severity;
    public String name;
    public String date;

    public Pic(){


    }

    public Pic(String path_leaf, String path_seg, float severity, String name, String date) {
        this.path_leaf = path_leaf;
        this.path_seg = path_seg;
        this.severity = severity;
        this.name = name;
        this.date = date;

    }

    public int length(){
        return date.length();
    }



    @Override
    public String toString() {
        return "Pic{" +
                "path_leaf='" + path_leaf + '\'' +
                ", path_seg='" + path_seg + '\'' +
                ", severity=" + severity +
                '}';
    }
}

File: app/src/main/java/com/rokid/cxrmsamples/data/Pic.kt
package com.rokid.cxrmsamples.data

data class Pic(
    val pathSeg: String,
    val pathFull: String?,
    val percent: Float,
    var name: String,
    var date: String,
    val createdAt: Long = System.currentTimeMillis(),
    val timingInfo: String? = null,
    val modelName: String? = null
)

fun String.toEnglishTimingInfo(): String = split(';')
    .map { it.trim() }
    .filter { it.isNotEmpty() }
    .filterNot {
        it.startsWith("Stage3") || it.startsWith("Stage 3", ignoreCase = true)
    }
    .joinToString("; ")
    .replace("模型推理(总)", "Inference total")
    .replace("Stage1粗分割", "Stage 1 coarse segmentation")
    .replace("Stage2精细分割", "Stage 2 fine segmentation")
    .replace("解码图片", "Decode image")
    .replace("缩放图片", "Resize image")
    .replace("生成名称", "Generate name")
    .replace("自动保存", "Auto-save")
    .replace("绘制叠加", "Render overlay")
    .replace("推理总计", "Inference total")
    .replace("总计", "Total")
